### PR TITLE
Modified as per review comments to split memorypressure config.

### PR DESCRIPTION
**Pull request by neeraj-deshpande**
Added two extra config variables: webprocesslimit, networkprocesslimit
Also added separate build flag for memory pressure APIs instead of using WebKit version check.

### CODE REVIEW

**Review by modeveci — APPROVED** [review]
LGTM

File: WebKitBrowser/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,9 @@ set(PLUGIN_WEBKITBROWSER_TRANSPARENT "false" CACHE STRING "Set transparency")
 set(PLUGIN_WEBKITBROWSER_STARTURL "about:blank" CACHE STRING "Initial URL for WebKitBrowser plugin")
 set(PLUGIN_WEBKITBROWSER_USERAGENT "" CACHE STRING "User agent string")
 set(PLUGIN_WEBKITBROWSER_MEMORYPROFILE "512m" CACHE STRING "Memory Profile")
-set(PLUGIN_WEBKITBROWSER_MEMORYPRESSURE "databaseprocess:50m,networkprocess:100m,webprocess:300m,rpcprocess:50m" CACHE STRING "Memory Pressure")
+set(PLUGIN_WEBKITBROWSER_MEMORYPRESSURE "" CACHE STRING "Memory Pressure")
+set(PLUGIN_WEBKITBROWSER_MEMORYPRESSURE_WEBPROCESSLIMIT "" CACHE STRING "Memory Pressure Webprocess Limit")
```
Comment (MFransen69) [inline WebKitBrowser/CMakeLists.txt:58]: Just to make sure, I discussed with Ozgur, better not to have a default in the new situation but use webkits one. So let's keep it like this

File: WebKitBrowser/WebKitBrowser.conf.in [hunk 2]
```diff
@@ -26,6 +26,8 @@ configuration.add("msebuffers", "@PLUGIN_WEBKITBROWSER_MSEBUFFERS@")
 configuration.add("thunderdecryptorpreference", "@PLUGIN_WEBKITBROWSER_THUNDER_DECRYPTOR_PREFERENCE@")
 configuration.add("memoryprofile", "@PLUGIN_WEBKITBROWSER_MEMORYPROFILE@")
 configuration.add("memorypressure", "@PLUGIN_WEBKITBROWSER_MEMORYPRESSURE@")
```
Comment (MFransen69) [inline WebKitBrowser/WebKitBrowser.conf.in:28]: let also add a "if(WEBKIT_MEMORY_PRESSURE_API)" (but of course one that is correct for python :) ). This makes clear you use one or the other (so not only rely of them being empty; this better explains how these options should be used

File: WebKitBrowser/WebKitBrowser.config [hunk 3]
```diff
@@ -41,7 +41,15 @@ map()
     kv(msebuffers ${PLUGIN_WEBKITBROWSER_MSEBUFFERS})
     kv(thunderdecryptorpreference ${PLUGIN_WEBKITBROWSER_THUNDER_DECRYPTOR_PREFERENCE})
     kv(memoryprofile ${PLUGIN_WEBKITBROWSER_MEMORYPROFILE})
-    kv(memorypressure ${PLUGIN_WEBKITBROWSER_MEMORYPRESSURE})
+    if(PLUGIN_WEBKITBROWSER_MEMORYPRESSURE)
```
Comment (MFransen69) [inline WebKitBrowser/WebKitBrowser.config:44]: Like in the WebKitBrowser.conf.in let's also use  "if(WEBKIT_MEMORY_PRESSURE_API)" here to make more clear that only either the old or new can be used but not both

File: WebKitBrowser/WebKitImplementation.cpp [hunk 4]
```diff
@@ -573,6 +577,8 @@ static GSourceFuncs _handlerIntervention =
             Core::JSON::Boolean ThunderDecryptorPreference;
             Core::JSON::String MemoryProfile;
             Core::JSON::String MemoryPressure;
+            Core::JSON::DecUInt32 WebProcessLimit;
```
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:580]: Let's also use  #ifdef WEBKIT_MEMORY_PRESSURE_API here. Not all three members are needed at the same time. Either the old is supported or the new one, but not both at the same time. This makes the code more efficient and also makes that concept more clear

File: WebKitBrowser/WebKitImplementation.cpp [hunk 5]
```diff
@@ -2167,7 +2173,26 @@ static GSourceFuncs _handlerIntervention =
...
+                    webkit_memory_pressure_settings_set_memory_limit(memoryPressureSettings, _config.NetworkProcessLimit.Value());
+                    webkit_website_data_manager_set_memory_pressure_settings(memoryPressureSettings);
+                    webkit_memory_pressure_settings_free(memoryPressureSettings);
+                }
+
+                WebKitWebsiteDataManager* websiteDataManager;
+                if (_config.WebProcessLimit.IsSet() == true) {
+                    WebKitMemoryPressureSettings* memoryPressureSettings = webkit_memory_pressure_settings_new();
+                    webkit_memory_pressure_settings_set_memory_limit(memoryPressureSettings, _config.WebProcessLimit.Value());
+                    websiteDataManager = webkit_website_data_manager_new("local-storage-directory", wpeStoragePath, "disk-cache-directory", wpeDiskCachePath, "memory-pressure-settings", memoryPressureSettings, nullptr);
```
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2188]: Ozgur will check if we only have the webkit_website_data_manager_new api. Perhaps we can split the directory settings from the memory settings and make the code a little more clear here

**Review by MFransen69 — CHANGES_REQUESTED** [review]

**Comment by modeveci** [comment]
Pseudo code for flow of APIs can be something like this:
1. config NetworkProcessLimit to #WebKitMemoryPressureSettings.
2. #WebkitMemoryPressureSettings instance to webkit_website_data_manager_set_memory_pressure_settings
3. Create a new data manager for instance websiteDataManager
4. Create a new webcontext instance by using websiteDataManager 
wkContext = webkit_web_context_new_with_website_data_manager(websiteDataManager); 
5. config WebProcessLimit to #WebKitMemoryPressureSettings.
6. #WebKitMemoryPressureSettings instance to the  created webcontext wkContext
7. g_obset_set(GOBJECT(wkContext),"memory-pressure-settings",#WebKitMemoryPressureSettings)

File: WebKitBrowser/CMakeLists.txt (hunk 1 above)
Comment (neeraj-deshpande) [reply WebKitBrowser/CMakeLists.txt:58]: done.

File: WebKitBrowser/WebKitBrowser.conf.in (hunk 2 above)
Comment (neeraj-deshpande) [reply WebKitBrowser/WebKitBrowser.conf.in]: done.

File: WebKitBrowser/WebKitBrowser.config (hunk 3 above)
Comment (neeraj-deshpande) [reply WebKitBrowser/WebKitBrowser.config]: done.

File: WebKitBrowser/WebKitImplementation.cpp (hunk 4 above)
Comment (neeraj-deshpande) [reply WebKitBrowser/WebKitImplementation.cpp:586]: done.

File: WebKitBrowser/WebKitImplementation.cpp (hunk 5 above)
Comment (neeraj-deshpande) [reply WebKitBrowser/WebKitImplementation.cpp]: Discussed with Ozgur.

**Review by MFransen69 — APPROVED** [review]

**Review by modeveci — APPROVED** [review]
Looks good to me. Thank you!

**Comment by modeveci** [comment]
This PR came to master branch by syncing with rdkservices. Closing

**Pull request closed without merging**